x-pack/plugins/observability_ai_assistant/server/functions/context.ts

```diff
@@ -77,49 +63,94 @@ export function registerRecallFunction({
         required: ['queries', 'categories'],
       } as const,
     },
-    async ({ arguments: { queries, categories }, messages, connectorId }, signal) => {
-      const systemMessage = messages.find((message) => message.message.role === MessageRole.System);
+    async ({ arguments: args, messages, connectorId, appContexts }, signal) => {
+      const { queries, categories } = args;
 
-      if (!systemMessage) {
-        throw new Error('No system message found');
-      }
+      async function getContext() {
+        const systemMessage = messages.find(
+          (message) => message.message.role === MessageRole.System
+        );
 
-      const userMessage = last(
-        messages.filter((message) => message.message.role === MessageRole.User)
-      );
+        const screenDescription = appContexts.map((context) => context.description).join('\n\n');
 
-      const nonEmptyQueries = compact(queries);
+        const content = { screen_description: screenDescription, learnings: [] };
 
-      const queriesOrUserPrompt = nonEmptyQueries.length
-        ? nonEmptyQueries
-        : compact([userMessage?.message.content]);
+        if (!isKnowledgeBaseAvailable) {
+          return { content };
+        }
 
-      const suggestions = await retrieveSuggestions({
-        userMessage,
-        client,
-        categories,
-        queries: queriesOrUserPrompt,
-      });
+        if (!systemMessage) {
+          throw new Error('No system message found');
+        }
+
+        const userMessage = last(
+          messages.filter((message) => message.message.role === MessageRole.User)
+        );
+
+        const nonEmptyQueries = compact(queries);
+
+        const queriesOrUserPrompt = nonEmptyQueries.length
+          ? nonEmptyQueries
+          : compact([userMessage?.message.content]);
+
+        queriesOrUserPrompt.push(screenDescription);
+
+        const suggestions = await retrieveSuggestions({
+          userMessage,
+          client,
+          categories,
+          queries: queriesOrUserPrompt,
+        });
+
+        if (suggestions.length === 0) {
+          return {
+            content,
+          };
+        }
+
+        const relevantDocuments = await scoreSuggestions({
+          suggestions,
+          queries: queriesOrUserPrompt,
+          messages,
+          client,
+          connectorId,
+          signal,
+          logger: resources.logger,
+        });
 
-      if (suggestions.length === 0) {
         return {
-          content: [] as unknown as Serializable,
+          content: { ...content, learnings: relevantDocuments as unknown as Serializable },
```

Comment:
Do we need it nested under `content`?

```suggestion
          ...content, learnings: relevantDocuments as unknown as Serializable,
```

Reply:
yes - we have two properties, `content` and `data`. Both are persisted to the conversation, but only `content` gets sent over to the LLM.